syntax = "proto3";

package order;

import "google/protobuf/timestamp.proto";

// Order service definition
service OrderService {
  // Place a new order
  rpc PlaceOrder (PlaceOrderRequest) returns (OrderResponse) {}
}

// Request message for placing an order
message PlaceOrderRequest {
  int32 customer_id = 1;
  repeated OrderItem items = 2;
}

// Message representing an item in the order
message OrderItem {
  int32 product_id = 1;
  int32 quantity = 2;
}

// Response message after placing an order
message OrderResponse {
  int32 id = 1;
  int32 customer_id = 2;
  double total_amount = 3;
  string status = 4;
  google.protobuf.Timestamp created_at = 5;
}
